te_dp/src/te_stats_collector.cpp

```diff
@@ -991,8 +988,6 @@ void stat_collector_thread(const char* tcp_socket_to_connect, vector<int> &activ
 
 cleanup:
     json_object_put(metrics);
-    for (auto it=metrics_obj_to_clear.begin(); it!=metrics_obj_to_clear.end(); it++)
-        json_object_put(*it);
```

Comment:
We were double freeing and that was leading to `malloc_consolidate` crash in te-stats-collector thereby killing te_dp process. The upgrade of json-c library may be the culprit,  the code  causing the issue was added as part of mem-leak issue, but looks like the new version of `json_object_object_add` frees the object that was used as val which need not be freed later
```
JSON_EXPORT int json_object_object_add	(	struct json_object * 	obj,
const char * 	key,
struct json_object * 	val 
)		
Add an object field to a json_object of type json_type_object

The reference count of val will not be incremented, in effect transferring ownership that object to obj, and thus val will be freed when obj is. (i.e. through json_object_put(obj))
```
https://json-c.github.io/json-c/json-c-current-release/doc/html/json__object_8h.html#a27bd808a022251059a43f1f6370441cd